h5c/vic/src/vic-webapp/src/app/shared/components/vch-operations/vch-operations.component.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2017 VMware, Inc. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+import {Component, Input, OnInit} from '@angular/core';
+import {VchComponentBase} from '../vch-component-base';
+import {VchOperationsView, VchViewKeys} from '../../../interfaces/vch';
+import {Observable} from 'rxjs/Observable';
+import {FormBuilder, Validators} from '@angular/forms';
+import {CreateVchWizardService} from '../../../create-vch-wizard/create-vch-wizard.service';
+import {I18nService} from '../../i18n.service';
+
+@Component({
+  selector: 'vic-vch-operations',
+  templateUrl: './vch-operations.component.html',
+  styleUrls: ['./vch-operations.component.scss']
+})
+export class VchOperationsComponent extends VchComponentBase implements OnInit {
+
+  @Input()
+  model: VchOperationsView;
+
+  protected readonly apiModelKey: VchViewKeys = 'operations';
+  protected readonly initialModel: VchOperationsView = {
+    opsUser: '',
+    opsPassword: '',
+    opsGrantPerms: false
+  };
+
+  constructor(protected formBuilder: FormBuilder,
+              protected createWzService: CreateVchWizardService,
+              public i18n: I18nService) {
+    super(formBuilder, createWzService);
+    this.updateCurrentForm(this.initialModel);
+  }
+
+  ngOnInit() {
+    super.ngOnInit();
+  }
+
+  onCommit(): Observable<{ [key: string]: VchOperationsView }> {
+    return Observable.of({[this.apiModelKey]: this.model});
+  }
+
+  protected updateCurrentForm(model: VchOperationsView) {
+    this.form = this.formBuilder.group({
```

Comment:
why not use patchValue to update the form instead of rebuilding it entirely?

this.form.patchValue({
  opsUser: [model.opsUser, Validators.required],
  ...
});

Reply:
Actually this method is not updating the form, the method is initializing the form that is why I can not use patch. My first aproach when I started working on the reconfigure tasks was to update the form, but after several changes the current method functionallity is to initialize the form .
I've refactored the method name in order to be in sync with the current method funcitonallity. Thanks for the tip :)